### PR TITLE
Migrate to AWS S3

### PR DESCRIPTION
**Pull request by mikestonecodes**
Something like this should be good for uploading the docs to s3. I don't think environment variables will work inside .travis.yml , so you can encrypt the data. see https://stackoverflow.com/questions/33735992/travis-ci-using-repository-environment-variables-in-travis-yml 

Fixes: #30

### CODE REVIEW

**Comment by mikestonecodes** [comment]
https://docs.travis-ci.com/user/encryption-keys/

**Comment by eswarasai** [comment]
@mikestonecodes -- ENV variables to work inside Travis CI. We're already using it across all our repos i.e., `website`, `dApp`, `cms`. 
Ref: https://github.com/MARKETProtocol/website/blob/develop/.travis.yml#L13

I'd appreciate if you can remove the deployment to `gh-pages` as well as add both `dev.docs.marketprotocol.io` and `docs.marketprotocol.io` deploy providers similar to the configuration of `website`, either me or @pelsasser can take care of Travis CI setup. Sounds good?

**Comment by mikestonecodes** [comment]
Good call on ENV variables. I'm a bit confused here. I think everything can be done within travis so not sure what I should do if I am not doing travis. gh-pages seems to only be deployed when deploy.sh is run, which is currently outside of travis. I'm not sure what should be uploaded to dev.docs.marketprotocol.io vs docs.marketprotocol.io.  There seems to just be a build folder that is generated by calling `bundle exec middleman build` and there is no 'develop' branch.

**Comment by rojosnow** [comment]
@mikestonecodes @eswarasai Would you guys connect this to an issue for tracking? Otherwise, the work doesn't show up in the iteration.

File: .travis.yml [hunk 1]
```diff
@@ -8,3 +8,13 @@ rvm:
 
 cache: bundler
 script: bundle exec middleman build
+deploy:
+  provider: s3
+  access_key_id:$AWS_ACCESS_KEY
```
Comment (eswarasai) [inline .travis.yml]: @mikestonecodes -- There needs to be a space here I guess before `$AWS_ACCESS_KEY`?

File: .travis.yml [hunk 2]
```diff
@@ -8,3 +8,13 @@ rvm:
 
 cache: bundler
 script: bundle exec middleman build
+deploy:
+  provider: s3
+  access_key_id: $AWS_ACCESS_KEY
```
Comment (eswarasai) [inline .travis.yml]: @mikestonecodes -- If you can update this block with the below one, the PR is good to be merged and tested. Let us know once done. Thanks!

```
access_key_id: $AWS_ACCESS_KEY_ID
secret_access_key: $AWS_SECRET_ACCESS_KEY
bucket: $AWS_BUCKET_NAME
```

**Comment by eswarasai** [comment]
@pelsasser -- This is good to be merged now!

**Pull request merged**